### ISSUE TITLE
Provisioning multiple datasources results in unauthorized error

### ISSUE THREAD

**Issue by Nkmol** [grafana/grafana#28511]
**What happened**:
When testing the second data source (http://localhost:3000/datasources/edit/2/), I get an unauthorized (403) error.

Please see the below to view the HAR text.

**What you expected to happen**:
The error is dependent on the number of datasources in the config file. Running them standalone makes them work. I would expect to provisioning file to support multiple datasources.

**How to reproduce it (as minimally and precisely as possible)**:

Add the following `docker-compose`:

```
version: '3.7'

services:
  influxdb:
    image: influxdb:1.7.8
    environment:
      - INFLUXDB_ADMIN_USER=admin
      - INFLUXDB_ADMIN_PASSWORD=admin
      - INFLUXDB_DB=telegraf
      - INFLUXDB_USER=user
      - INFLUXDB_USER_PASSWORD=user
      - INFLUXDB_REPORTING_DISABLED=true
  grafana:
    image: grafana/grafana:7.2.0
    environment: 
      - GF_SECURITY_ADMIN_PASSWORD=admin
      - INFLUXDB_ADMIN_USER=admin
      - INFLUXDB_ADMIN_PASSWORD=admin
      - INFLUXDB_DATABASE=test
    volumes:
      - ./datasource.yaml:/etc/grafana/provisioning/datasources/datasource.yaml
    ports:
      - 3000:3000
```

The mounted YAML data source:

```
apiVersion: 1

datasources:
  - name: influxdb-telegraf
    type: influxdb
    access: proxy
    database: telegraf
    user: admin
    url: http://influxdb:8086
    default: true
    secureJsonData:
      password: admin

  - name: influxdb-telegraf-extra
    type: influxdb
    access: proxy
    database: telegraf
    user: admin
    url: http://influxdb:8086
    jsonData:
      timeInterval: "10s"
    secureJsonData:
      password: admin
```

**Anything else we need to know?**:

You can view the following network traffic (HAR):

```
{
  "log": {
    "version": "1.2",
    "creator": {
      "name": "Firefox",
      "version": "82.0"
    },
    "browser": {
      "name": "Firefox",
      "version": "82.0"
    },
    "pages": [
      {
        "startedDateTime": "2020-10-23T18:09:46.391+02:00",
        "id": "page_4",
        "title": "Loading: Settings - Grafana",
        "pageTimings": {
          "onContentLoad": -1,
          "onLoad": -1
        }
      }
    ],
    "entries": [
      {
        "pageref": "page_4",
        "startedDateTime": "2020-10-23T18:09:46.391+02:00",
        "request": {
          "bodySize": 0,
          "method": "GET",
          "url": "http://localhost:3000/api/datasources/proxy/2/query?db=telegraf&q=SHOW%20RETENTION%20POLICIES%20on%20%22telegraf%22&epoch=ms",
          "httpVersion": "HTTP/1.1",
          "headers": [
            {
              "name": "Host",
              "value": "localhost:3000"
            },
            {
              "name": "User-Agent",
              "value": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.16; rv:82.0) Gecko/20100101 Firefox/82.0"
            },
            {
              "name": "Accept",
              "value": "application/json, text/plain, */*"
            },
            {
              "name": "Accept-Language",
              "value": "nl,en-US;q=0.7,en;q=0.3"
            },
            {
              "name": "Accept-Encoding",
              "value": "gzip, deflate"
            },
            {
              "name": "Referer",
              "value": "http://localhost:3000/datasources/edit/2/"
            },
            {
              "name": "x-grafana-nocache",
              "value": "true"
            },
            {
              "name": "x-grafana-org-id",
              "value": "1"
            },
            {
              "name": "DNT",
              "value": "1"
            },
            {
              "name": "Authorization",
              "value": "Basic YWRtaW46UjdjNXFCa3ZYMGwx"
            },
            {
              "name": "Connection",
              "value": "keep-alive"
            },
            {
              "name": "Cookie",
              "value": "GS_FLOW_CONTROL=GS_CFLOW_-ddee9e4:1754aa646d6:7e03; io=uZoQduHvZLVnt9mXAAAC; grafana_session=862e78d9a925e0d608966c2c55daa853"
            }
          ],
          "cookies": [
            {
              "name": "GS_FLOW_CONTROL",
              "value": "GS_CFLOW_-ddee9e4:1754aa646d6:7e03"
            },
            {
              "name": "io",
              "value": "uZoQduHvZLVnt9mXAAAC"
            },
            {
              "name": "grafana_session",
              "value": "862e78d9a925e0d608966c2c55daa853"
            }
          ],
          "queryString": [
            {
              "name": "db",
              "value": "telegraf"
            },
            {
              "name": "q",
              "value": "SHOW RETENTION POLICIES on \"telegraf\""
            },
            {
              "name": "epoch",
              "value": "ms"
            }
          ],
          "headersSize": 667
        },
        "response": {
          "status": 401,
          "statusText": "Unauthorized",
          "httpVersion": "HTTP/1.1",
          "headers": [
            {
              "name": "Content-Type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "X-Content-Type-Options",
              "value": "nosniff"
            },
            {
              "name": "X-Frame-Options",
              "value": "deny"
            },
            {
              "name": "X-Xss-Protection",
              "value": "1; mode=block"
            },
            {
              "name": "Date",
              "value": "Fri, 23 Oct 2020 16:09:46 GMT"
            },
            {
              "name": "Content-Length",
              "value": "42"
            }
          ],
          "cookies": [],
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 42,
            "text": "{\"message\":\"Invalid username or password\"}"
          },
          "redirectURL": "",
          "headersSize": 222,
          "bodySize": 264
        },
        "cache": {},
        "timings": {
          "blocked": 1,
          "dns": 0,
          "connect": 0,
          "ssl": 0,
          "send": 0,
          "wait": 21,
          "receive": 0
        },
        "time": 22,
        "_securityState": "insecure",
        "serverIPAddress": "127.0.0.1",
        "connection": "3000"
      },
      {
        "pageref": "page_4",
        "startedDateTime": "2020-10-23T18:09:46.425+02:00",
        "request": {
          "bodySize": 0,
          "method": "GET",
          "url": "http://localhost:3000/api/login/ping",
          "httpVersion": "HTTP/1.1",
          "headers": [
            {
              "name": "Host",
              "value": "localhost:3000"
            },
            {
              "name": "User-Agent",
              "value": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.16; rv:82.0) Gecko/20100101 Firefox/82.0"
            },
            {
              "name": "Accept",
              "value": "application/json, text/plain, */*"
            },
            {
              "name": "Accept-Language",
              "value": "nl,en-US;q=0.7,en;q=0.3"
            },
            {
              "name": "Accept-Encoding",
              "value": "gzip, deflate"
            },
            {
              "name": "Referer",
              "value": "http://localhost:3000/datasources/edit/2/"
            },
            {
              "name": "x-grafana-nocache",
              "value": "true"
            },
            {
              "name": "x-grafana-org-id",
              "value": "1"
            },
            {
              "name": "DNT",
              "value": "1"
            },
            {
              "name": "Connection",
              "value": "keep-alive"
            },
            {
              "name": "Cookie",
              "value": "GS_FLOW_CONTROL=GS_CFLOW_-ddee9e4:1754aa646d6:7e03; io=uZoQduHvZLVnt9mXAAAC; grafana_session=862e78d9a925e0d608966c2c55daa853"
            }
          ],
          "cookies": [
            {
              "name": "GS_FLOW_CONTROL",
              "value": "GS_CFLOW_-ddee9e4:1754aa646d6:7e03"
            },
            {
              "name": "io",
              "value": "uZoQduHvZLVnt9mXAAAC"
            },
            {
              "name": "grafana_session",
              "value": "862e78d9a925e0d608966c2c55daa853"
            }
          ],
          "queryString": [],
          "headersSize": 532
        },
        "response": {
          "status": 200,
          "statusText": "OK",
          "httpVersion": "HTTP/1.1",
          "headers": [
            {
              "name": "Cache-Control",
              "value": "no-cache"
            },
            {
              "name": "Content-Encoding",
              "value": "gzip"
            },
            {
              "name": "Content-Type",
              "value": "application/json"
            },
            {
              "name": "Expires",
              "value": "-1"
            },
            {
              "name": "Pragma",
              "value": "no-cache"
            },
            {
              "name": "Vary",
              "value": "Accept-Encoding"
            },
            {
              "name": "X-Content-Type-Options",
              "value": "nosniff"
            },
            {
              "name": "X-Frame-Options",
              "value": "deny"
            },
            {
              "name": "X-Xss-Protection",
              "value": "1; mode=block"
            },
            {
              "name": "Date",
              "value": "Fri, 23 Oct 2020 16:09:46 GMT"
            },
            {
              "name": "Content-Length",
              "value": "37"
            }
          ],
          "cookies": [],
          "content": {
            "mimeType": "application/json",
            "size": 9,
            "text": "Logged in"
          },
          "redirectURL": "",
          "headersSize": 300,
          "bodySize": 337
        },
        "cache": {},
        "timings": {
          "blocked": 1,
          "dns": 0,
          "connect": 0,
          "ssl": 0,
          "send": 0,
          "wait": 6,
          "receive": 0
        },
        "time": 7,
        "_securityState": "insecure",
        "serverIPAddress": "127.0.0.1",
        "connection": "3000"
      },
      {
        "pageref": "page_4",
        "startedDateTime": "2020-10-23T18:09:46.441+02:00",
        "request": {
          "bodySize": 0,
          "method": "GET",
          "url": "http://localhost:3000/api/datasources/proxy/2/query?db=telegraf&q=SHOW%20RETENTION%20POLICIES%20on%20%22telegraf%22&epoch=ms",
          "httpVersion": "HTTP/1.1",
          "headers": [
            {
              "name": "Host",
              "value": "localhost:3000"
            },
            {
              "name": "User-Agent",
              "value": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.16; rv:82.0) Gecko/20100101 Firefox/82.0"
            },
            {
              "name": "Accept",
              "value": "application/json, text/plain, */*"
            },
            {
              "name": "Accept-Language",
              "value": "nl,en-US;q=0.7,en;q=0.3"
            },
            {
              "name": "Accept-Encoding",
              "value": "gzip, deflate"
            },
            {
              "name": "Referer",
              "value": "http://localhost:3000/datasources/edit/2/"
            },
            {
              "name": "x-grafana-nocache",
              "value": "true"
            },
            {
              "name": "x-grafana-org-id",
              "value": "1"
            },
            {
              "name": "DNT",
              "value": "1"
            },
            {
              "name": "Authorization",
              "value": "Basic YWRtaW46UjdjNXFCa3ZYMGwx"
            },
            {
              "name": "Connection",
              "value": "keep-alive"
            },
            {
              "name": "Cookie",
              "value": "GS_FLOW_CONTROL=GS_CFLOW_-ddee9e4:1754aa646d6:7e03; io=uZoQduHvZLVnt9mXAAAC; grafana_session=862e78d9a925e0d608966c2c55daa853"
            }
          ],
          "cookies": [
            {
              "name": "GS_FLOW_CONTROL",
              "value": "GS_CFLOW_-ddee9e4:1754aa646d6:7e03"
            },
            {
              "name": "io",
              "value": "uZoQduHvZLVnt9mXAAAC"
            },
            {
              "name": "grafana_session",
              "value": "862e78d9a925e0d608966c2c55daa853"
            }
          ],
          "queryString": [
            {
              "name": "db",
              "value": "telegraf"
            },
            {
              "name": "q",
              "value": "SHOW RETENTION POLICIES on \"telegraf\""
            },
            {
              "name": "epoch",
              "value": "ms"
            }
          ],
          "headersSize": 667
        },
        "response": {
          "status": 401,
          "statusText": "Unauthorized",
          "httpVersion": "HTTP/1.1",
          "headers": [
            {
              "name": "Content-Type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "X-Content-Type-Options",
              "value": "nosniff"
            },
            {
              "name": "X-Frame-Options",
              "value": "deny"
            },
            {
              "name": "X-Xss-Protection",
              "value": "1; mode=block"
            },
            {
              "name": "Date",
              "value": "Fri, 23 Oct 2020 16:09:46 GMT"
            },
            {
              "name": "Content-Length",
              "value": "42"
            }
          ],
          "cookies": [],
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 42,
            "text": "{\"message\":\"Invalid username or password\"}"
          },
          "redirectURL": "",
          "headersSize": 222,
          "bodySize": 264
        },
        "cache": {},
        "timings": {
          "blocked": 1,
          "dns": 0,
          "connect": 0,
          "ssl": 0,
          "send": 0,
          "wait": 3,
          "receive": 0
        },
        "time": 4,
        "_securityState": "insecure",
        "serverIPAddress": "127.0.0.1",
        "connection": "3000"
      }
    ]
  }
}
```

You can view this here: http://gitgrimbo.github.io/harviewer/master/

**Environment**:
- Grafana version: Both 7.2.0 and 6.5.2
- Data source type & version: Influxdb
- OS Grafana is installed on: Docker
- User OS & Browser: Firefox MacOs

**Comment by hugohaggmark** [comment]
Hi @Nkmol I added ports and changed the url in datasources.yaml as follows:

__docker-compose.yaml:__
```yaml
version: "3.7"

services:
  influxdb:
    image: influxdb:1.7.8
    ports:
      - "8086:8086"
    environment:
      - INFLUXDB_ADMIN_USER=admin
      - INFLUXDB_ADMIN_PASSWORD=admin
      - INFLUXDB_DB=telegraf
      - INFLUXDB_USER=user
      - INFLUXDB_USER_PASSWORD=user
      - INFLUXDB_REPORTING_DISABLED=true
  grafana:
    image: grafana/grafana:7.2.0
    environment:
      - GF_SECURITY_ADMIN_PASSWORD=admin
      - INFLUXDB_ADMIN_USER=admin
      - INFLUXDB_ADMIN_PASSWORD=admin
      - INFLUXDB_DATABASE=test
    volumes:
      - ./datasource.yaml:/etc/grafana/provisioning/datasources/datasource.yaml
    ports:
      - 3000:3000
```

__datasources.yaml:__
```yaml
apiVersion: 1

datasources:
  - name: influxdb-telegraf
    type: influxdb
    access: proxy
    database: telegraf
    user: admin
    url: http://host.docker.internal:8086
    default: true
    secureJsonData:
      password: admin

  - name: influxdb-telegraf-extra
    type: influxdb
    access: proxy
    database: telegraf
    user: admin
    url: http://host.docker.internal:8086
    jsonData:
      timeInterval: "10s"
    secureJsonData:
      password: admin
```

I couldn't repro your issue with this setup so please let me know if you still experience any issue?

**Comment by Nkmol** [comment]
@hugohaggmark Thanks for the quick response!

Apologies, but I provided a wrong `datasource.yml`. I was actually using the direct docker network service name: "http://influxdb:8086". I edited this in my original post.

I initially tried the exact setup you have provided, but still found that the second ("influxdb-telegraf-extra") did not work and was again resulting in the unauthorized (401) HTTP error.

----
**tl;dr: ** 'Refreshing' Firefox through the `about:config` tab solved the issue.

Weirdly enough, I contacted a colleague to test this with me and it also worked for him. So down the rabbit hole I went:
- Synced the `docker-comopse` and `docker` version
- Synced browser versions 
- Disabled all browser add-ons
- Removed local caching

I then tried incognito mode, which seemed to work once out of multiple tries. So this also does not seem to be 100% reliable. After this, I went on and reset (or 'refreshed' as it is called in Firefox) through the `about:config` tab. This seemed to have solved all my issues.

Sadly, I am unsure what setting would have caused this. But maybe these findings would help anyone also encountering these problems. As a side note, while testing I also had problems logging in; resulting in the same unauthorized error. I believe this symptom is related to this issue.

----

Thanks for the time and I am happy I finally solved this problem 👍
